types/src/write_set.rs

```diff
@@ -42,6 +45,26 @@ impl std::fmt::Debug for WriteOp {
     }
 }
 
+impl CanonicalSerialize for WriteOp {
+    fn serialize(&self, serializer: &mut impl CanonicalSerializer) -> Result<()> {
+        serializer.encode_bool(self.is_value())?;
```

Comment:
We are treating this enum differently from others. Should we consider unifying them or is there specific reason not to?

Reply:
The assumption being that this enum is fixed in types, but I am happy to go either way. Given that I already had a funny feeling about this, I'll switch to the common enum process even if it adds 3 more bytes ... unless you feel otherwise.

Reply:
Yup, agreed. Let's go with the general way, so that we don't need to spec them differently.